subql_wika/src/mappings/eventHandlers.ts

```diff
@@ -0,0 +1,163 @@
+
+    // Data Plugins
+    private postgres: PluginPostgres ;
+    private neo4j: PluginNeo4j ;
+    private elastic: PluginElasticSearch ;
+    private emails: PluginEmails ;
+
+    // Logger
+    private log ;
+
+    /**
+    * Constructor relies on env vars
+    * for the list of env vars, see the docs of each Plugin, or the docker-compose file
+    */
+    constructor() {
+        // Instantiate data plugins
+        this.postgres = new PluginPostgres() ;
+        this.neo4j = new PluginNeo4j() ;
+        this.elastic = new PluginElasticSearch() ;
+        this.emails = new PluginEmails() ;
+        // If there's a global logger variable, use it,
+        // otherwise, forward to console
+        try {
+            this.log = logger ;
+        } catch (e) {
+            this.log = console ;
+        }
+    }
+
+    /**
+    * Handle to the Postgres plugin
+    */
+    getPluginPostgres() {
+        return this.postgres ;
+    }
+
+    /**
+    * Sends an email if EmailsPlugin is enabled
+    * Plus, logs the error
+    * @param e - The error to report
+    */
+    logError(e) {
+        if (this.emails.isEnabled()) {
+            this.emails.sendError(e.toString()) ;
+        }
+        this.log.error(e);
+    }
+
+    /**
+    * Fetches a webpage and saves the metadata to Postgres and Elastic Search
+    * (when plugins are enabled)
+    * @param url - The url string
+    */
+    async updateMetadata(url) {
+        const metadata = await fetchMetadata(url) ;
+        if (metadata) {
+            if (this.postgres.isSyncEnabled()) {
+                this.postgres.newMetadataRecord(url, metadata) ;
+            } else {
+                this.log.warn('postgres is disabled')
+            }
+            if (this.elastic.isSyncEnabled()) {
+                await this.elastic.postUrl(url, metadata) ;
+            } else {
+                this.log.warn('elastic is disabled')
+            }
+        }
+    }
+
+    /**
+    * Processes a LikeEvent
+    * Updates the metadata in Postgres and Elastic Search
+    * Records the event in Postgres
+    * Creates or updates the LIKE relationship in Neo4J
+    * @param event - Polkadot API Event, containing user, url and numLikes in its data field
+    */
+    async handleLikeEvent(event) {
+        const eventData = event.event.data ;
+        const blockId = event.extrinsic.block.block.header.hash.toString() ;
+        const blockNum = event.extrinsic.block.block.header.number.toNumber();
+        const eventId = blockNum + '/' + event.idx ;
+        const user = eventData[0].toString() ;
+        const url = eventData[1].toHuman().toString() ;
+        const numLikes = Number(eventData[2]) ;
+        this.log.info('handleLikeEvent : '+eventId+' : '+user+' : '+url+' : '+numLikes) ;
+
+        // Update metadata
+        this.updateMetadata(url) ;
+
+        // Main record
+        if (this.postgres.isSyncEnabled()) {
+            this.postgres.newLikeEvent(eventId, blockNum, url, user, numLikes);
```

Comment:
Is there a reason you don't `await` here?

Reply:
Nope, just forgot. Fixed.